hopsworks-common/src/main/java/io/hops/hopsworks/common/hive/HiveController.java

```diff
@@ -0,0 +1,184 @@
+package io.hops.hopsworks.common.hive;
+
+import io.hops.hopsworks.common.dao.dataset.Dataset;
+import io.hops.hopsworks.common.dao.dataset.DatasetFacade;
+import io.hops.hopsworks.common.dao.dataset.DatasetType;
+import io.hops.hopsworks.common.dao.hdfs.inode.Inode;
+import io.hops.hopsworks.common.dao.hdfs.inode.InodeFacade;
+import io.hops.hopsworks.common.dao.hdfsUser.HdfsUsers;
+import io.hops.hopsworks.common.dao.project.Project;
+import io.hops.hopsworks.common.dao.user.Users;
+import io.hops.hopsworks.common.hdfs.DistributedFileSystemOps;
+import io.hops.hopsworks.common.hdfs.HdfsUsersController;
+import io.hops.hopsworks.common.util.BaseHadoopClientsService;
+import io.hops.hopsworks.common.util.Settings;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ejb.EJB;
+import javax.ejb.Stateless;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.Statement;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Stateless(name = "HiveController")
+public class HiveController {
+
+  @EJB
+  private Settings settings;
+  @EJB
+  private HdfsUsersController hdfsUsersBean;
+  @EJB
+  private InodeFacade inodeFacade;
+  @EJB
+  private DatasetFacade datasetFacade;
+  @EJB
+  private BaseHadoopClientsService bhcs;
+
+
+  private final static String driver = "org.apache.hive.jdbc.HiveDriver";
+  private final static Logger logger = Logger.getLogger(HiveController.class.getName());
+
+  private Connection conn;
+  private String jdbcString = null;
+
+  @PostConstruct
+  public void init() {
+    try {
+      // Load Hive JDBC Driver
+      Class.forName(driver);
+
+      // Create connection url
+      String hiveEndpoint = settings.getHiveServerHostName(false);
```

Comment:
@jimdowling @SirOibaf @kouzant Wouldn't it be better to use a Datasource than doing all that manually in here?

Reply:
I see that @SirOibaf  is creating and dropping DBs here. That isn't part of JPA. So, i think it's reasonable to use JDBC here. The important thing is that there are no hard-coded params here, and everything is coming from Settings.java, so that is ok by me.